src/python/platforms/android/adb.py

```diff
@@ -571,7 +571,7 @@ def run_command(cmd,
   if (output in [
       DEVICE_HANG_STRING, DEVICE_OFFLINE_STRING,
       device_not_found_string_with_serial
-  ]):
+  ]) and not environment.is_android_emulator():
```

Comment:
i think this needs to be moved up,

if not recover or environment.is_android_emulator():
on line 564
since basically we don't want this to trigger on emulator?

Reply:
I agree. I've made the change.